Source/Scene/Cesium3DTilesetTraversal.js

```diff
@@ -1,856 +1,643 @@
-                    allReady = allReady && child.contentAvailable;
-                } else {
-                    allReady = allReady && this.internalDFS.execute(child);
-                }
+    function selectDesiredTile(tileset, tile, frameState) {
+        if (!skipLevelOfDetail(tileset)) {
+            if (tile.contentAvailable) {
+                // The tile can be selected right away and does not require traverseAndSelect
+                selectTile(tileset, tile, frameState);
+                tile._finalResolution = true;
             }
+            return;
         }
 
-        if (allReady) {
-            return children;
+        // If this tile is not loaded attempt to select its ancestor instead
+        var loadedTile = tile.contentAvailable ? tile : tile._ancestorWithContentAvailable;
+        if (defined(loadedTile)) {
+            // Tiles will actually be selected in traverseAndSelect
+            loadedTile._shouldSelect = true;
+        } else {
+            // If no ancestors are ready traverse down and select tiles to minimize empty regions.
+            // This happens often for immediatelyLoadDesiredLevelOfDetail where parent tiles are not necessarily loaded before zooming out.
+            selectDescendants(tileset, tile, frameState);
         }
+    }
 
-        return emptyArray;
-    };
+    function visitTile(tileset, tile, frameState) {
+        ++tileset._statistics.visited;
+        tile._visitedFrame = frameState.frameNumber;
+    }
 
-    function baseUpdateAndCheckChildren(tileset, tile, baseScreenSpaceError, frameState) {
-        if (hasAdditiveContent(tile)) {
-            tileset._desiredTiles.push(tile);
+    function touchTile(tileset, tile, frameState) {
+        if (tile._touchedFrame === frameState.frameNumber) {
+            // Prevents another pass from touching the frame again
+            return;
         }
+        tileset._cache.touch(tile);
+        tile._touchedFrame = frameState.frameNumber;
+    }
 
-        // Stop traversal on the subtree since it will be destroyed
-        if (tile.hasTilesetContent && tile.contentExpired) {
-            return false;
+    function getPriority(tileset, tile) {
+        // If skipLevelOfDetail is off try to load child tiles as soon as possible so that their parent can refine sooner.
+        // Additive tiles are prioritized by distance because it subjectively looks better.
+        // Replacement tiles are prioritized by screen space error.
+        // A tileset that has both additive and replacement tiles may not prioritize tiles as effectively since SSE and distance
+        // are different types of values. Maybe all priorities need to be normalized to 0-1 range.
+        var parent = tile.parent;
+        var replace = tile.refine === Cesium3DTileRefine.REPLACE;
+        var add = tile.refine === Cesium3DTileRefine.ADD;
+        if (add) {
+            return tile._distanceToCamera;
+        } else if (replace) {
```

Comment:
Do we want to return some priority as a default, or leave the priority `undefined`?

Reply:
Also you don't need the `else if` because of the early return above.